server/container/guice/cassandra-rabbitmq-ldap-guice/src/test/java/org/apache/james/CassandraRabbitMQLdapJamesServerTest.java

```diff
@@ -37,54 +38,84 @@
-        .extension(new LdapTestExtention())
-        .extension(new SwiftBlobStoreExtension())
-        .server(configuration -> GuiceJamesServer.forConfiguration(configuration)
-            .combineWith(CassandraRabbitMQLdapJamesServerMain.MODULES)
-            .overrideWith(new TestJMAPServerModule(LIMIT_TO_10_MESSAGES))
-            .overrideWith(DOMAIN_LIST_CONFIGURATION_MODULE))
-        .build();
-
-    @Test
-    void userFromLdapShouldLoginViaImapProtocol(GuiceJamesServer server) throws IOException {
-        imapClient.connect(JAMES_SERVER_HOST, server.getProbe(ImapGuiceProbe.class).getImapPort());
-
-        assertThat(imapClient.login(JAMES_USER, PASSWORD)).isTrue();
+
+    interface MailsShouldBeWellReceived {
+        @RegisterExtension
+        IMAPMessageReader imapMessageReader = new IMAPMessageReader();
+        SMTPMessageSender messageSender = new SMTPMessageSender(Domain.LOCALHOST.asString());
+
+        @Test
+        default void mailsShouldBeWellReceived(GuiceJamesServer server) throws Exception {
+            messageSender.connect(JAMES_SERVER_HOST, server.getProbe(SmtpGuiceProbe.class).getSmtpPort())
+                .sendMessage("bob@any.com", JAMES_USER + "@localhost");
+
+            calmlyAwait.until(() -> server.getProbe(SpoolerProbe.class).processingFinished());
+
+            imapMessageReader.connect(JAMES_SERVER_HOST, server.getProbe(ImapGuiceProbe.class).getImapPort())
+                .login(JAMES_USER, PASSWORD)
+                .select("INBOX")
+                .awaitMessage(calmlyAwait);
+        }
     }
 
-    @Test
-    void mailsShouldBeWellReceivedBeforeFirstUserConnectionWithLdap(GuiceJamesServer server) throws Exception {
-        messageSender.connect(JAMES_SERVER_HOST, server.getProbe(SmtpGuiceProbe.class).getSmtpPort())
-            .sendMessage("bob@any.com", JAMES_USER + "@localhost");
+    interface UserFromLdapShouldLogin {
 
-        calmlyAwait.until(() -> server.getProbe(SpoolerProbe.class).processingFinished());
+        @Test
+        default void userFromLdapShouldLoginViaImapProtocol(GuiceJamesServer server) throws IOException {
+            IMAPClient imapClient = new IMAPClient();
+            imapClient.connect(JAMES_SERVER_HOST, server.getProbe(ImapGuiceProbe.class).getImapPort());
+
+            assertThat(imapClient.login(JAMES_USER, PASSWORD)).isTrue();
+        }
+    }
+
+    interface ContratSuite extends JmapJamesServerContract, MailsShouldBeWellReceived, UserFromLdapShouldLogin {}
+
+    @Nested
+    @TestInstance(TestInstance.Lifecycle.PER_CLASS)
+    class WithSwift implements ContratSuite {
+        @RegisterExtension
+        JamesServerExtension testExtension = new JamesServerExtensionBuilder()
+            .extension(new EmbeddedElasticSearchExtension())
+            .extension(new CassandraExtension())
+            .extension(new RabbitMQExtension())
+            .extension(new LdapTestExtention())
+            .extension(new SwiftBlobStoreExtension())
```

Comment:
actually, you still need a binding statement to bind SwiftBlobStore to `BlobStore`, otherwise. James will use CassandraBlobStore as the default. [See](https://github.com/linagora/james-project/blob/735ef66c38f0bf594e420d8b0822979753ccbad7/server/container/guice/cassandra-rabbitmq-guice/src/main/java/org/apache/james/modules/blobstore/BlobStoreChoosingModule.java) from PR #1917 
to get the way to deal with this problem. An example is in #1911 : https://github.com/linagora/james-project/blob/3d7b0bb89cd2112ec11fb730b2c7abfe10e9d232/server/container/guice/cassandra-rabbitmq-guice/src/test/java/org/apache/james/CassandraRabbitMQSwiftJmapTestRule.java#L71

Reply:
We can bind **the choosing configuration** to the one with Swift